### PR TITLE
convert tests to use make

### PR DESCRIPTION
**Pull request by pweil-**
Use make targets for unit, int, and e2e tests so they can be maintained in a single place.

### CODE REVIEW

**Comment by pweil-** [comment]
@smarterclayton @danmcp PTAL.  

Reference: https://github.com/openshift/origin/issues/867#issuecomment-72792176

File: lib/vagrant-openshift/action/run_openshift3_tests.rb [hunk 1]
```diff
@@ -55,26 +55,38 @@ def run_tests(env, cmds, as_root=true)
...
-          end
+          cmd_env = []
+          build_targets = ["make"]
 
           if @options[:all]
-            cmds << 'hack/test-integration.sh'
-            cmds << 'hack/test-cmd.sh'
+            cmd_env << 'ARTIFACT_DIR=/tmp/origin/e2e/artifacts'
+            cmd_env << 'LOG_DIR=/tmp/origin/e2e/logs'
+            build_targets << 'test'
```
Comment (smarterclayton) [inline lib/vagrant-openshift/action/run_openshift3_tests.rb]: Hrm - test also runs build, but we want to be testing the thing that build-release generated.  We may need to define a new make target, or figure out how to skip the build.

Comment (pweil-) [inline lib/vagrant-openshift/action/run_openshift3_tests.rb]: Updated the make file to add a SKIP_BUILD flag that controls if the build is done or not.  https://github.com/openshift/origin/pull/886

**Comment by pweil-** [comment]
Local testing:

test target (unit tests only)
```
[pweil@localhost origin]$ vagrant test-openshift3
***************************************************
Running  make check...
hack/test-go.sh  
... snip...
ok  	github.com/openshift/origin/test/templates	0.038s

real	2m25.184s
user	1m12.314s
sys	1m20.365s
Finished  make check
***************************************************
```

test all target (requires images to be present already, utilizes skip build flag)
```
[pweil@localhost origin]$ vagrant build-openshift3 --images
Running command '
... snip ...

[pweil@localhost origin]$ vagrant test-openshift3 --all
***************************************************
Running ARTIFACT_DIR=/tmp/origin/e2e/artifacts LOG_DIR=/tmp/origin/e2e/logs make test SKIP_BUILD=true...
build is being skipped
hack/test-go.sh 
... snip ...
ok  	github.com/openshift/origin/test/templates	1.171s	coverage: 0.0% of statements
hack/test-cmd.sh
... snip ...
Complete
hack/test-integration.sh 
etcd: {"action":"get","node":{"key":"/","dir":true}}

Integration test cases ...

ok  	github.com/openshift/origin/test/integration	7.726s

Complete
hack/test-integration-docker.sh 
etcd: {"action":"get","node":{"key":"/","dir":true}}

Docker integration test cases ...

ok  	github.com/openshift/origin/test/integration	26.840s

Complete
hack/test-end-to-end.sh
... snip ...
```

**Pull request merged**